doc/developer_guide/conf.py

```diff
@@ -37,9 +42,26 @@
 extensions = ['sphinx.ext.autodoc', 'sphinx.ext.doctest',
               'sphinx.ext.intersphinx', 'sphinx.ext.coverage',
               'sphinx.ext.imgmath', 'sphinx.ext.viewcode',
-              'sphinxcontrib.bibtex']
+              'sphinxcontrib.bibtex', 'apilinks']
 bibtex_bibfiles = ['../bibliography/references.bib']
 
+# GITHUB_PR_NUMBER is set in .github/workflows/python-package.yml when
+# running the document-testing job.
+github_pr_num = os.environ.get('GITHUB_PR_NUMBER')
+
+if github_pr_num:
+    # Generate links to a webserver running locally if we're generating
+    # documentation as part of the CI run on GHA.
+    usr_guide_base = 'http://0.0.0.0:8000/user_guide/_build/html/'
+    ref_guide_base = (
+        'http://0.0.0.0:8000/reference_guide/build/html/autogenerated/')
+else:
```

Comment:
I'm not sure what you mean here. At the moment I have configured GHA to set an environment variable so that we know we're building the docs inside the test environment. If that variable is not set then we assume we building the docs 'for real' and link to RTD. Whether we choose to link to 'latest' or 'stable' is a separate question. Are you suggesting (as I think you are) that we should link to 'stable' versions of the docs when building 'stable' versions of the docs? I'll have to investigate whether there's some way of finding that out on RTD.

Reply:
I found that there is an appropriate environment variable so I've implemented this.